src/audio/pipeline.c

```diff
@@ -627,13 +630,17 @@ static int pipeline_comp_trigger(struct comp_dev *current,
 					    ppl_data->start->pipeline);
 	int err;
 
+	printf("pipeline_comp_trigger(), current->comp.id = %u, dir = %u\n",
+		    dev_comp_id(current), dir);
 	pipe_cl_dbg("pipeline_comp_trigger(), current->comp.id = %u, dir = %u",
 		    dev_comp_id(current), dir);
 
+	printf("is_single_ppl=%d, is_same_sched=%d\n", is_single_ppl, is_same_sched);
```

Comment:
I found in pipeline.c, pipeline_comp_trigger() takes "is_same_sched" into regard, and then pipeline_comp_trigger_sched_comp() to list current pipeline into ctx. So using DAI_ADD_SCHED should be correct.

According to output log:
pipeline_comp_trigger(), current->comp.id = 0, dir = 0
is_single_ppl=1, is_same_sched=1
pipeline_comp_trigger(), current->comp.id = 1, dir = 0
is_single_ppl=1, is_same_sched=1
pipeline_comp_trigger(), current->comp.id = 7, dir = 0
is_single_ppl=0, is_same_sched=1
pipeline_comp_trigger(), current->comp.id = 4, dir = 0
is_single_ppl=1, is_same_sched=1

comp.id 7 is the second DAI, in theory it should be listed.